src/geometry/java/us/ihmc/euclid/geometry/tools/EuclidGeometryPolygonTools.java

```diff
@@ -236,12 +218,20 @@ public static int inPlaceGiftWrapConvexHull2D(List<? extends Point2DReadOnly> ve
                return numberOfVertices;
          }
 
-         while ((lastHullVertexIndex >= 1 && isCandidatePointCollinearInOppositeDirectionWithTheFirstPoint(lastHullVertex,
+         while ( candidateIndex < numberOfVertices && lastHullVertexIndex >= 1 && isCandidatePointCollinearInOppositeDirectionWithTheFirstPoint(lastHullVertex,
                                                                                                            vertices.get(lastHullVertexIndex - 1),
                                                                                                            candidateVertex,
-                                                                                                           EPSILON)))
+                                                                                                           EPSILON))
          { // The next candidate vertex isn't valid because it's collinear, but may be valid for a future point, so we want to keep it in scope.
             candidateIndex++;
+
+            // This is exceptional case when skpping over the duplicate or collinear points, this ran out of valid candidates but still tried to access the next one.
+            // Forcely, increasing index number for this.
+            if (candidateIndex >= numberOfVertices)
+            {
+               return lastHullVertexIndex +1;
```

Comment:
There should be a space after the `+`.
```suggestion
               return lastHullVertexIndex + 1;
```